src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Tree/Dependencies/Snapshot/Filters/IAddDependencyContext.cs

```diff
@@ -0,0 +1,55 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Immutable;
+
+namespace Microsoft.VisualStudio.ProjectSystem.VS.Tree.Dependencies.Snapshot.Filters
+{
+    /// <summary>
+    /// Context used by <see cref="IDependenciesSnapshotFilter"/> implementations when filtering
+    /// a dependency that is being added to a <see cref="DependenciesSnapshot"/> by
+    /// <see cref="DependenciesSnapshot.FromChanges"/>.
+    /// </summary>
+    internal interface IAddDependencyContext
```

Comment:
Is this following a particular pattern? I've not seen it before.

Reply:
See https://github.com/dotnet/project-system/pull/4296#discussion_r235331280